pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -114,7 +114,19 @@ func (step *commitImageStep) execute(ctx *postExecutorStepContext) error {
 
 	ctx.labels = createLabelsForResultingImage(step.builder, step.docker, step.image)
 
-	ctx.imageID, err = commitContainer(step.docker, ctx.containerID, cmd, user, step.builder.config.Tag, step.builder.env, ctx.labels)
+	// Set the image entrypoint back to its original value on commit, the running
+	// container has "env" as its entrypoint and we don't want to commit that.
+	entrypoint, err := step.docker.GetImageEntrypoint(step.image)
+	// If the image has no explicit entrypoint, set it to an empty array
+	// so we don't default to leaving the entrypoint as "env" upon commit.
+	if entrypoint == nil {
+		entrypoint = []string{}
+	}
+	if err != nil {
+		return fmt.Errorf("Couldn't get entrypoint of %q image: %v", step.image, err)
+	}
+
+	ctx.imageID, err = commitContainer(step.docker, ctx.containerID, cmd, user, step.builder.config.Tag, step.builder.env, entrypoint, ctx.labels)
```

Comment:
> "see an about"?

I mean that I'm expecting that there is no more than 5 lines of code to be added.